### PR TITLE
Remove drand48() usage

### LINKED ISSUE #1907: Remove drand48() usage
Per Coverity:
>  "drand48()" should not be used for security related applications, as linear congruential algorithms are too easy to break.

### PR DESCRIPTION
**Pull request by v1siuol**
Fix #1907

### CODE REVIEW

**Comment by v1siuol** [comment]
Minors in jtest:
We have duplicated `#include <stdlib.h>`, not sure if this is intentional. 

`MAX_FILE_ARGUMENTS` is not used in this file any more. It was introduced in e8ad2978176e62da929c4343db89b26f100b1353. And the usage was removed in 25cd96dcf882afb4d4306982c17181b27bb8ae54.

Please let me know if you have any questions. Thanks in advance.

**Comment by masaori335** [comment]
Thanks for the cleanup! Looks reasonable.

> We have duplicated #include <stdlib.h>, not sure if this is intentional.
>
> MAX_FILE_ARGUMENTS is not used in this file any more. It was introduced in e8ad297. And the usage was removed in 25cd96d.

It looks like these are other cleanups we need to do. Let's do it on other PRs.

@bneradt please fire up CI jobs for this.

**Comment by bneradt** [comment]
[approve ci]

**Comment by bneradt** [comment]
@v1siuol : Thank you for the PR. It looks like the clang-format build is failing. You can see the output from the run here:
https://ci.trafficserver.apache.org/job/Github_Builds/job/clang-format/476/console

This can most easily be addressed by running `make clang-format` in your local build tree. That will apply the fixes in place, allowing you to commit them and re-push your branch. The CI should re-run automatically then. @ mention me if it doesn't and I'll kick it off again.

**Review by masaori335 — APPROVED** [review]
Look reasonable

**Comment by v1siuol** [comment]
Thanks @masaori335. I notice that there are a few more `drand48()` usages in the following files.  Should I remove them as well?
```
trafficserver/example/plugins/c-api/thread_pool/test/SDKTest/psi_server.c
trafficserver/iocore/aio/test_AIO.cc
trafficserver/iocore/cache/CacheDir.cc
trafficserver/iocore/cache/CacheTest.cc
trafficserver/plugins/cache_promote/policy.cc
trafficserver/proxy/hdrs/unit_tests/test_Hdrs.cc
trafficserver/src/tscore/unit_tests/test_MMH.cc
```

**Comment by masaori335** [comment]
Yes, `ts::Random` is introduced to replace it ( #7859 ). Either is fine to continue on this PR or make another PR.

File: example/plugins/c-api/thread_pool/test/SDKTest/psi_server.c [hunk 1]
```diff
@@ -68,7 +69,7 @@ int
 generate_psibility()
 {
   double rand;
-  rand = drand48();
+  rand = ts::Random::drandom();
```
Comment (v1siuol) [inline example/plugins/c-api/thread_pool/test/SDKTest/psi_server.c:72]: not sure if I should edit this c file (I don't know how to run it either

Comment (masaori335) [reply example/plugins/c-api/thread_pool/test/SDKTest/psi_server.c:72]: This file looks abandoned for a long time. We can just replace it for now.
https://github.com/apache/trafficserver/issues/8561

**Review by masaori335 — CHANGES_REQUESTED** [review]
Please roll back the changes in the c file. Let's figure out the issue independently.

**Review by masaori335 — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**